Purchases/Caching/RCDeviceCache.m

```diff
@@ -101,15 +101,16 @@ - (void)cacheAppUserID:(NSString *)appUserID {
     [self.userDefaults setObject:appUserID forKey:RCAppUserDefaultsKey];
 }
 
-- (void)clearCachesForAppUserID:(NSString *)appUserID {
+- (void)clearCachesForAppUserID:(NSString *)oldAppUserID andSaveNewUserID:(NSString *)newUserID {
     @synchronized (self) {
         [self.userDefaults removeObjectForKey:RCLegacyGeneratedAppUserDefaultsKey];
-        [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
```

Comment:
Why was this removed?

Reply:
it was replaced by `self.cacheAppUserID`, so you never have a nil appUserID, which is an invalid state